classes/submission/SubmissionDAO.inc.php

```diff
@@ -527,18 +530,23 @@ function getBySubEditorId($contextId, $subEditorId = null, $includeDeclined = tr
 	 * Get all unpublished submissions for a user.
 	 * @param $userId int
 	 * @param $contextId int optional
+	 * @param $id int|null optional Filter by id.
 	 * @param $title string? Optional title search string to limit results
 	 * @param $stageId int? Optional stage ID to limit results
 	 * @param $sectionId int? Optional section ID to limit results
 	 * @param $rangeInfo DBResultRange optional
 	 * @return array Submissions
 	 */
-	function getUnpublishedByUserId($userId, $contextId = null, $title = null, $stageId = null, $sectionId = null, $rangeInfo = null) {
+	function getUnpublishedByUserId($userId, $contextId = null, $id = null, $title = null, $stageId = null, $sectionId = null, $rangeInfo = null) {
 		$params = array_merge(
 			$this->getFetchParameters(),
 			array((int) ROLE_ID_AUTHOR, (int) $userId)
 		);
-		if ($title) $params[] = '%' . $title . '%';
+		if ($id) $params[] = (int) $id;
+		if ($title) {
+			$params[] = 'title';
```

Comment:
@asmecher the previous implementation broke the sql, and therefore the search at the submissions->My Queue -> My Authored grid. 
I haven't found any Issue for that, but if there is any, maybe we could close it.